fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -182,8 +182,13 @@ def fit(self, X, y, **kwargs):
     def predict(self, X):
         """Provide a prediction for the given input data.
 
-        Note that this is non-deterministic, due to the nature of the
-        exponentiated gradient algorithm.
+        Chooses the prediction label randomly based on the probabilities from `_pmf_predict`,
```

Comment:
Here I would only say that the predictions are randomized, so the output might be different on the same data and that this behavior is intended. We could provide more detail either in the user guide or, for now, in the `Notes` section (see [numpy standard](https://numpydoc.readthedocs.io/en/latest/format.html#sections)), e.g., there we can say that a fitted `ExponentiatedGradient` has `predictors_` and `weights_` and on each data point we pick a predictor randomly and apply it. I don't think it makes sense to mention anything about `_pmf_predict` here--it's such a low level detail, that it just belongs into a comment in the code.

Reply:
Thanks! Addressed this.